repository/src/main/java/org/apache/atlas/repository/store/graph/v1/DeleteHandlerV1.java

```diff
@@ -505,7 +505,6 @@ public List<AtlasVertex> addTagPropagation(AtlasVertex classificationVertex, Lis
                 RequestContext.get().endMetricRecord(countMetricRecorder);
             }
         }
-        if (ret.isEmpty()) return null;
```

Comment:
even we remove null return here, line 2796 in EntityGraphMapper will return null for processChunkedPropagation if the collection is empty. Why not keep the null logic and check for null to exit early in propagateClassification?

my preference is still to initialize as null and only allocate new ArrayList when necessary